Descent3/AImain.cpp

```diff
@@ -1625,7 +1625,7 @@ const char *const Ai_movement_subtype_walking_strings[MAX_AI_INIT_MOVEMENT_SUBTY
 #define AI_MAX_MELEE_RANGE 5.0f
 
 int AI_NumRendered;
-int AI_RenderedList[MAX_OBJECTS];
+std::array<int, MAX_OBJECTS> AI_RenderedList;
```

Comment:
As long as we're doing these conversions, let's default-initialize the elements in the arrays:
```suggestion
std::array<int, MAX_OBJECTS> AI_RenderedList{};
```
Both `T[N]` and `std::array<T, N>` default-initialize only the _array object_, not the array _elements_. IE, this gives an indeterminate value:
```
std::array<int, 4> arr;
int val = arr[0];
// val is indeterminate
```

Reply:
I mean... OK but I was expecting most of these to eventually be changed to `std::vector`.

Reply:
That would be preferable, definitely.